### ISSUE TITLE
unexpected behaviour in numpy implementation

### ISSUE THREAD

**Issue by iamVarunAnand** [prlz77/orthoreg#2]
Hey, 
I have tried to implement this regularizer on a toy dataset using numpy. But my data points end up shrinking in value rather than moving away from each other. I would appreciate if you can point out the error in my code.

`
t = np.random.randn(16, 2)

for i in range(20):
    # normalize the activations
    tn = t / (np.linalg.norm(t, axis = -1, keepdims = True) + 1e-6)

    grad = np.matmul(tn, tn.T)
    grad = (grad * lambd) / (grad + np.exp(lambd))

    grad = grad * (1 - np.eye(grad.shape[0], dtype = grad.dtype))
    grad = np.matmul(grad, tn)

    # calculate the coeff
    coeff = -1 * beta * lr

    # adjust the activations
    t = t + (grad * coeff)    

`

**Comment by prlz77** [comment]
Hi @iamVarunAnand thanks for the interest in our work! I just compared it with the pytorch version and it seems alright. In fact it is possible that if you directly apply this function without an opposing force (i.e. backprop gradient from the output), you get shrinkage in the representations. One possible solution is that you save the norm of each vector before adding the gradient, and recover it afterwards.
